test/client/js/libs/handlebars/parse.js

```diff
@@ -312,6 +325,35 @@ describe("parse Handlebars helper", () => {
 		expect(actual).to.deep.equal(expected);
 	});
 
+	// Emoji
+	[{
+		name: "in text",
+		input: "Hello💬",
+		expected: 'Hello<span class="emoji">💬</span>',
+	}, {
+		name: "with modifiers",
+		input: "🤷‍♀️",
+		expected: '<span class="emoji">🤷‍♀️</span>',
+	}, {
+		name: "wrapped in style",
+		input: "Super \x034💚 green!",
+		expected: 'Super <span class="emoji"><span class="irc-fg4">💚</span></span><span class="irc-fg4"> green!</span>',
+	}, {
+		name: "wrapped in URLs",
+		input: "https://i.❤️.thelounge.chat",
+		// FIXME: Emoji in text should be `<span class="emoji">❤️</span>`
```

Comment:
This was "broken" when overlapping parts were fixed to support nicknames in messages, so I wouldn't consider this a bug.

Reply:
I really think the class should be applied though in the text part of the link/channel though. It's odd that it looks smaller, and if we add things to the `emoji` element (like a tooltip with name as it was mentioned some time ago), it will be even weirder.

Reply:
It's going to be a pain in the ass for that work correctly (you don't want spans in `href`), so I don't know if it's worth the effort at all.

Reply:
Right, I don't deny that, but still a bug nonetheless :)